.buildkite/pipeline.yml

```diff
@@ -0,0 +1,138 @@
+        key: unit-test
+        command: ".buildkite/scripts/unit_test.sh"
+        agents:
+          provider: "gcp"
+        artifact_paths:
+          - build/*.xml
+
+      - label: ":smartbear-testexecute: Run integration tests"
+        key: int-test
+        command: ".buildkite/scripts/integration_test.sh"
+        agents:
+          provider: "gcp"
+        artifact_paths:
+          - build/*.xml
+
+      - label: ":junit: Junit annotate"
+        plugins:
+          - junit-annotate#v2.4.1:
+              artifacts: "build/*.xml"
+        agents:
+          provider: "gcp"
+        depends_on:
+          - step: "unit-test"
+            allow_failure: true
+          - step: "int-test"
+            allow_failure: true
+
+      - label: "E2E Test"
+        key: "e2e-test"
+        command: ".buildkite/scripts/e2e_test.sh"
+        agents:
+          provider: "gcp"
+        depends_on:
+          - step: "unit-test"
+            allow_failure: false
+          - step: "int-test"
+            allow_failure: false
+
+      - label: ":gcloud: Cloud e2e Test"
+        key: "cloud-e2e-test"
+        command: ".buildkite/scripts/cloud_e2e_test.sh"
+        agents:
+          provider: "gcp"
+        depends_on:
+          - step: "unit-test"
+            allow_failure: false
+          - step: "int-test"
+            allow_failure: false
+          - step: "e2e-test"
+            allow_failure: false
+    depends_on: "check"
+
+  - label: ":docker: Publish docker image"
+    key: "publish"
+    command: ".buildkite/scripts/build_push_docker_image.sh"
+    if: "build.env('BUILDKITE_PULL_REQUEST') == 'false' && build.env('BUILDKITE_BRANCH') == 'main'"
+    agents:
+      provider: "gcp"
+    depends_on:
+      - step: "tests"
+        allow_failure: false
+
+  - label: ":argo: Update fleet image tag for fleet-controller using gpctl"
+    async: true
+    branches: main
+    trigger: gpctl-promote
+    build:
+      env:
+        SERVICE_COMMIT_HASH: ${BUILDKITE_COMMIT:0:12}
+        REMOTE_SERVICE_CONFIG: https://raw.githubusercontent.com/elastic/serverless-gitops/main/gen/gpctl/fleet/config.yaml
+    depends_on:
+      - step: "publish"
+
+  - label: ":gcloud: Release test"
+    key: "release-test"
+    command: ".buildkite/scripts/release_test.sh"
+    agents:
+      provider: "gcp"
+    depends_on:
+      - step: "tests"
+        allow_failure: false
+
+  - label: ":jenkins: Release - Package Registry Distribution"
+    key: "release-package-registry"
+    trigger: "package-registry-release-package-registry-distribution"
+    build:
+      branch: "main"
+      meta_data:
+        DOCKER_TAG: "${BUILDKITE_TAG}"
+    if: "build.env('BUILDKITE_TAG') != ''"
+    depends_on:
+      - step: "release-test"
+        allow_failure: false
+
+  - trigger: "fleet-server-package-mbp"
+    label: ":esbuild: Downstream - Package"
+    key: "downstream-package"
+    if: "build.env('BUILDKITE_PULL_REQUEST') == 'false'"
```

Comment:
I think this needs to be udpated as in this PR: https://github.com/elastic/fleet-server/pull/2890